src/codeu/chat/server/BroadCastSystem.java

```diff
@@ -0,0 +1,148 @@
+package codeu.chat.server;
+
+import codeu.chat.common.*;
+import codeu.chat.util.Serializers;
+import codeu.chat.util.connections.Connection;
+import codeu.chat.util.Uuid;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+
+/**
+ * Created by rsharif on 3/17/17.
+ */
+
+
+public class BroadCastSystem {
+
+    private class ConversationMessageLink {
+
+        Uuid conversationUuid;
+        Message message;
+
+        ConversationMessageLink(Uuid conversationUuid, Message message) {
+            this.conversationUuid = conversationUuid;
+            this.message = message;
+        }
+
+    }
+
+    private ConcurrentHashMap<Integer, List<Connection>> conversationUsers;
+    private final BlockingQueue<ConversationMessageLink> messagesToBroadcast;
+    private final Thread broadCaster;
+
+    private class MessageBroadCaster implements Runnable {
+        @Override
+        public void run() {
+
+            try {
+                while (true) {
+                    broadCastMessage(messagesToBroadcast.take());
```

Comment:
So this exception is actually thrown when the blocking queue was interrupted while waiting for a message to be added. The issue is not with a specific message that failed to broadcast. 

If a failed attempt to broadcast a message happens, it will happen when sending to a specific user, so simply adding it back to queue would result in certain users getting the message more than once. 

I like the idea of adding multiple attempts to send a message because at the moment the connection is closed if there is a failed attempt to send a message. I will look into how to implement this.